crates/block-producer/src/server/mod.rs

```diff
@@ -1,62 +1,45 @@
-use std::{
-    collections::{BTreeMap, BTreeSet},
-    net::ToSocketAddrs,
-    sync::Arc,
-};
+use std::{net::ToSocketAddrs, sync::Arc};
 
-use miden_node_proto::{
-    domain::nullifiers,
-    generated::{
-        block_producer::api_server, requests::SubmitProvenTransactionRequest,
-        responses::SubmitProvenTransactionResponse, store::api_client as store_client,
-    },
+use miden_node_proto::generated::{
+    block_producer::api_server, requests::SubmitProvenTransactionRequest,
+    responses::SubmitProvenTransactionResponse, store::api_client as store_client,
 };
 use miden_node_utils::{
     errors::ApiError,
     formatting::{format_input_notes, format_output_notes},
 };
-use miden_objects::{
-    transaction::ProvenTransaction, utils::serde::Deserializable, MIN_PROOF_SECURITY_LEVEL,
-};
-use miden_tx::TransactionVerifier;
+use miden_objects::{transaction::ProvenTransaction, utils::serde::Deserializable};
 use tokio::{net::TcpListener, sync::Mutex};
 use tokio_stream::wrappers::TcpListenerStream;
 use tonic::Status;
 use tracing::{debug, info, instrument};
 
 use crate::{
-    batch_builder::{DefaultBatchBuilder, DefaultBatchBuilderOptions},
-    block_builder::DefaultBlockBuilder,
+    batch_builder::BatchProver,
+    block_builder::BlockProver,
     config::BlockProducerConfig,
     domain::transaction::AuthenticatedTransaction,
     errors::{AddTransactionError, VerifyTxError},
-    mempool::Mempool,
-    state_view::DefaultStateView,
+    mempool::{BlockNumber, Mempool},
     store::{DefaultStore, Store},
-    txqueue::{TransactionQueue, TransactionQueueOptions},
-    COMPONENT, SERVER_BATCH_SIZE, SERVER_BLOCK_FREQUENCY, SERVER_BUILD_BATCH_FREQUENCY,
-    SERVER_MAX_BATCHES_PER_BLOCK,
+    COMPONENT, SERVER_BATCH_SIZE, SERVER_MAX_BATCHES_PER_BLOCK, SERVER_MEMPOOL_STATE_RETENTION,
 };
 
-pub mod api;
-
-type Api = api::BlockProducerApi<
-    DefaultBatchBuilder<
-        DefaultStore,
-        DefaultBlockBuilder<DefaultStore, DefaultStateView<DefaultStore>>,
-    >,
-    DefaultStateView<DefaultStore>,
->;
-
 /// Represents an initialized block-producer component where the RPC connection is open,
 /// but not yet actively responding to requests.
 ///
 /// Separating the connection binding from the server spawning allows the caller to connect other
 /// components to the store without resorting to sleeps or other mechanisms to spawn dependent
 /// components.
 pub struct BlockProducer {
-    api_service: api_server::ApiServer<Api>,
-    listener: TcpListener,
+    batch_config: BatchProver,
+    block_config: BlockProver,
```

Comment:
The `_config` suffix here is a bit confusing. These fields are not just configs, right?

Reply:
Initially they were; I was hoping to have a single struct which contains just the configurable values, and then runtime specific things get passed into the `run` method e.g.

```rust
struct BlockProver {
    pub block_interval: Duration,
    pub simulated_proof_time: Range<Duration>,
    pub failure_rate: f32,
}

impl BlockProver {
    async fn run(&self, mempool, store)
}
```

which had the benefit that we could implement `Default` for it with the `const SERVER_XXX` values.

It didn't mesh well with the current flow in the node start command so I undid it.

I'll maybe streamline this again in a future PR; will fixup the naming thanks.